python/az/aro/azext_aro/_validators.py

```diff
@@ -287,6 +287,13 @@ def validate_version_format(namespace):
     if namespace.version is not None and not re.match(r'^[4-9]{1}\.[0-9]{1,2}\.[0-9]{1,2}$', namespace.version):
         raise InvalidArgumentValueError('--version is invalid')
 
+def validate_upgradeable_to_format(namespace):
+    if not namespace.upgradeable_to:
+        return
+    if not re.match(r'^[4-9]{1}\.[0-9]{1,2}\.[0-9]{1,2}$', namespace.upgradeable_to):
+        raise InvalidArgumentValueError('--upgradeable-to is invalid')
+    if namespace.client_secret is not None or namespace.client_id is not None:
```

Comment:
Should/Can we add this check in client_secret and client_id validations rather than adding here? Suggesting this because for other MIWI fields that validation is perfromed in the client_secret and client_id.

Reply:
I agree with this suggestion, this makes more sense than what I proposed previously (current implementation in this PR right now).